doc/specs/#140 - ZIP Support.md

```diff
@@ -0,0 +1,180 @@
+2) The archive contains only one supported installer (msix, msi, or exe).
+3) The archive contains more than one file.
+
+## Solution Design
+
+A new installation flow need to be created to execute different installation strategies regarding the three cases above.
+
+First thing first, the archive needs to be extracted in a specific path (TEMP folder can be a good candidate).
+
+Regarding case 1, installation flow check if TEMP folder contain a single binary and repack that binary
+into an MSI(X) package (including setting PATH env var). Then, MSI(X) installation flow can take control.
+
+Regarding case 2, installation flow check if TEMP folder contain an installer (.exe, .msi, .msix) and pass control
+to MSI(X) installation flow.
+
+Regarding case 3, installation flow check if TEMP folder does not contain an installer nor a single binary.
+This case is the complex one, example with 'vagrant' hierarchy:
+- root
+  - bin 
+    - vagrant.exe
+    - other.exe
+  - embedded
+    - bin
+    - some other folders
+
+This hierarchy can be packed into an MSI(X) installer, but how can we specify which binary to add 
+in PATH env var (in that case bin/vagrant.exe)?
+
+This specification take on consideration schema on version 1.1.0. 
+A schema update should be necessary to take all the cases in consideration. 
+
+Take a look into the following manifest examples
+
+### Case 1|2 YAML manifest
+
+```
+PackageIdentifier: Editor.App
+PackageVersion: 1.0
+MinimumOSVersion: 10.0.0.0
+InstallerType: zip
+Scope: ...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+ManifestType: installer
+ManifestVersion: 1.0.0
+```
+
+This is compliant with the actual schema v1.1.0
+
+### Case 2.1 YAML manifest
+
+If the archive contains an installer but not located at root folder (e.g. archive.zip/somefolder/installer.exe)
+
+```
+PackageIdentifier: Editor.App
+PackageVersion: 1.0
+MinimumOSVersion: 10.0.0.0
+InstallerType: zip
+Scope: ...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+  InstallerPath: somefolder/installer.exe
+ManifestType: installer
+ManifestVersion: 1.0.0
+```
+
+This is not compliant with the actual schema v1.1.0.
+```InstallerPath``` attribute should be added to schema.
+
+### Case 3 YAML manifest
+
+```
+PackageIdentifier: Editor.App
+PackageVersion: 1.0
+MinimumOSVersion: 10.0.0.0
+InstallerType: zip
+Scope: ...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+  BinaryPath: bin (e.g. bin folder where vagrant.exe is located in my example)
```

Comment:
Could this be simplified to "InstallerPath" as well, and have the cli determine if it is an installer or binary?

Reply:
Case 2 cover installer (.exe, .msi, etc). You're pointing case 3, which is for complex hierarchy that does not include installer, only binary with complex folders structure. The base idea is to implement all the cases. Nonetheless, "BInaryPath" maybe not a good name, any idea ?

Reply:
I’m saying that for case 2, you are suggesting that a key be added to define the path to the exe/msi/etc.

Why can’t that same key be used for case 3?

Reply:
I got confused you was quoting case 3. On case 3, BinaryPath key is for indicate to register a path after the zip extraction into user or system environment variables (need to be defined how to). 

Correct me if I'm wrong, but from my point of view regarding case 2 (excluding my example manifest 2.1). Case 2 assuming that the zip archive only contains one file, the installer. Case 2 also assuming that the installer will set the proper environment variables if needed (specifically $PATH) or not. that depend of the nature of the application and how editors package their applications. I don't know, how do you manage that kind of thing, winget-cli is agnostic regarding this aspect ?

To answer your question, case 3 does not handle installer file. That the job of case 2 and 2.1. Case 2 does not need to define InstallerPath key because, the archive only contains the installer at the root archive. On the other hand, case 2.1 zip archive maybe not contains only one file and maybe not located at root, so InstallerPath key is needed to indicate where to find the installer after the extraction and execute it.

To summarize, BinaryPath and InstallerPath are optional keys that depend on the context and they should be set properly by the package maintainer. To avoid messing up with these settings, updating the documentation and the helper script YAMLCreate.ps1 should be needed, in my opinion. For all cases, if the manifest is miss-configured the procedure should fail in an organic way. 

I don't know if I'm clear, my english is not perfect. Maybe I should clarify these aspects on the specification.